### PR TITLE
[OpenSCAD] Add 2 new commands: Scale and Resize, fix bug in Mirror me…

### PR DESCRIPTION
**Pull request by mwganson**
…sh feature

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, please confirm the following:

- [ ] Branch rebased on latest master `git pull --rebase upstream master`
- [ ] Unit tests confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [ ] Commit message is [well-written](https://chris.beams.io/posts/git-commit/)
- [ ] Commit message includes `issue #<id>` or `fixes #<id>` where `<id>` is the [associated MantisBT](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) issue id if one exists

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.19 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=34586).

---

### CODE REVIEW

**Comment by mwganson** [comment]
https://forum.freecadweb.org/viewtopic.php?f=3&t=49204&start=10

File: src/Mod/OpenSCAD/OpenSCADUtils.py [hunk 1]
```diff
@@ -301,14 +301,32 @@ def vec2householder(nv):
...
+    poly = mesh2polyhedron(msh)
+    vec_string = '['+str(vec.x)+','+str(vec.y)+','+str(vec.z)+']'
+    param = 'scale('+vec_string+')'
+    mi = callopenscadmeshstring('%s{%s}' % (param,''.join(poly)))
+    mi.flipNormals()
+    return mi
+
+def resizemesh(msh,vec):
+    """resizemesh(mesh,vector) where mesh is a mesh object and vector is a Base.Vector"""
+    from exportCSG import mesh2polyhedron
```
Comment (vocx-fc) [inline src/Mod/OpenSCAD/OpenSCADUtils.py:323]: You should add the import at the beginning of the file.

File: src/Mod/OpenSCAD/OpenSCADCommands.py [hunk 2]
```diff
@@ -188,6 +188,67 @@ def GetResources(self):
...
+                'MenuText': QtCore.QT_TRANSLATE_NOOP('OpenSCAD_ScaleMeshFeature', 'Scale Mesh Feature...'),
+                'ToolTip' : QtCore.QT_TRANSLATE_NOOP('OpenSCAD_ScaleMeshFeature', 'Create Scale Mesh Feature')}
+
+
+class ResizeMeshFeature:
+    def IsActive(self):
+        return FreeCADGui.Selection.countObjectsOfType('Mesh::Feature') > 0
+
+    def Activated(self):
+        import Part,OpenSCADFeatures,OpenSCADUtils
```
Comment (vocx-fc) [inline src/Mod/OpenSCAD/OpenSCADCommands.py:227]: Neither `Part` nor `OpenSCADFeatures` seem to be used, so why import them? Surely `OpenSCADUtils` was already imported at the beginning of the file, no?

File: src/Mod/OpenSCAD/OpenSCADCommands.py [hunk 3]
```diff
@@ -188,6 +188,67 @@ def GetResources(self):
...
+        selection=FreeCADGui.Selection.getSelectionEx()
+        for selobj in selection:
+            newobj=selobj.Document.addObject("Mesh::Feature",'scale')
+            newobj.Label='scale_%s' % selobj.Object.Label
+            msh=selobj.Object.Mesh
+            items=["[1;1;1]"]
+            item, ok = QtGui.QInputDialog.getItem(QtGui.QApplication.activeWindow(),u'Scale about which Axis?',u'Enter scaling value:',items,editable=True)
```
Comment (vocx-fc) [inline src/Mod/OpenSCAD/OpenSCADCommands.py:203]: No need for explicit Unicode strings, `u'Something'`.

**Comment by mwganson** [comment]
Thanks for reviewing.  I have implemented your suggestions.

**Pull request merged**